### DISCUSSION TITLE
Hyperlinks - work in Supporting Content but not Chat window

### DISCUSSION THREAD

**Discussion by IainD925** [microsoft/AI-Chat-App-Hack#136] [Q&A]
`<a href="https://www.w3schools.com/">Visit W3Schools.com!</a>`

When source docs contain hyperlinks (as per example) I'm seeing that within the 'Supporting content' window it is clickable but not clickable within the chat/ask response window.

Have tried various prompts, best I get is basic text of the Hyperlink.  Anyone know what's needed to enable?

**Update**: [looks to be a known issue](https://community.openai.com/t/custom-gpt-bug-hyperlinks-not-clickable/565499/43?page=2)

**Comment by pamelafox** [comment]
Is GPT actually responding with HTML? I'm surprised as it usually responds in either plain text or markdown. Can you screenshot where it responds with HTML?
Generally, it's a bad idea to render HTML that you haven't written yourself, so our React components default to escaping HTML inside them. There is a way to render HTML in React via dangerouslySetInnerHTML, so you could try that. A better approach is to get the GPT to return Markdown links and render the Markdown using a technique like this PR:
https://github.com/Azure-Samples/azure-search-openai-demo/pull/56

**Reply by IainD925** [reply]
Thanks @pamelafox - the hyperlinks are in my source docs (not generated).

I tried integrating the 4 files from the PR you cite however upon `azd deploy` the frontend would not render.  
Just a blank page.  Left it for 10 mins, restarted app, refreshed etc, no joy.

3/4 files are small updates but [app/frontend/package-lock.json](https://github.com/Azure-Samples/azure-search-openai-demo/pull/56/files#diff-95ac4dadc642a030d3f73ce6253b4f8e7f18b32c6dd82dfc40ee46521feddbf6) is a beast.
I tried manually merging & wow, was that painful.

Reverted the 4 files, deployed, page appeared as normal after 1 minute.  I appreciate the link but it's beyond me.

**Comment by IainD925** [comment]
Found a workaround, update the prompt to 

> Output EVERYTHING in the content below. Use markup for html.
